core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java

```diff
@@ -66,6 +65,16 @@
         return (PagedModel<EnvironmentDto>) environmentDtoPagedResourcesAssembler.toEmptyModel(environmentDtoPage, EnvironmentDto.class);
     }
 
+    @GetMapping("/list")
+    @PreAuthorize("hasPrivilege('ENVIRONMENTS_READ')")
+    public HalMultipleEmbeddedResource<EnvironmentDto> getAll() {
+        List<Environment> environments = environmentService.getAll();
+        return new HalMultipleEmbeddedResource<>(
+                environments.stream().filter(distinctByKey(Environment::getName))
+                        .map(environment -> environmentDtoResourceAssembler.toModel(environment))
+                        .collect(Collectors.toList()));
+    }
```

Comment:
can't you use environmentDtoService#getAll(Page.unpaged) instead?

Reply:
Done